### DISCUSSION TITLE
Unable to get a docker-based GitHub Packages registry_package event from a GitHub Actions workflow

### DISCUSSION THREAD

**Discussion by jeking3** [community/community#25639] [Actions]
<p>I added a workflow that publishes to GitHub Packages (docker) on the creation of a release.  Then I added another workflow to activate on <code>registry_package</code>, however that event is not firing at all.</p>
<p>I made a forkable project that demonstrates this problem.  I am trying to use <code>registry_package</code> to trigger a qualification test run on my release docker image… but it never fires.  I’m hoping that I’m just doing something silly, otherwise this looks like a problem at GitHub.</p>
<p><a href="https://github.com/jeking3/no-github-package-docker-registry-package-event" rel="nofollow noopener">https://github.com/jeking3/no-github-package-docker-registry-package-event</a></p>
<p>The only action that shows up is the “release” action which indeed publishes a package, however I never get a <code>registry_package</code> workflow event to trigger <code>qualify.yaml</code>.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>The root cause is that an action in a workflow run which act with GITHUB_TOKEN can’t trigger a new workflow run. In your workflow, you login github docker registry with GITHUB_TOKEN. This make the docker image push operation act with GITHUB_TOKEN. So your workflow on: registry_package will not be triggered. </p>
<p>Please create a Personal Access Token with write:packages scope and store its value to <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets#creating-encrypted-secrets" rel="nofollow noopener">secrets</a>. Then use this secret variable in docker login step:</p>
<p><img src="https://user-images.githubusercontent.com/24806478/181092312-66bdc476-3f8d-4149-aeb6-be1e55b6ae1a.png" alt="docker login.png" data-base62-sha1="mWTjRHAGLy3Wvxp8rowFm2ozm80" width="459" height="109"></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7c0fcbd8001398db393b9b9eb2e99af127fbb916.png" data-download-href="/uploads/short-url/hHv34XkoxJZJfZlCMYjwMaxXv2C.png?dl=1" title="registry_package.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181092313-849c7889-01fc-4f74-ba7a-d3feeba3baa1.png" alt="registry_package.png" data-base62-sha1="hHv34XkoxJZJfZlCMYjwMaxXv2C" width="690" height="108" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/7c0fcbd8001398db393b9b9eb2e99af127fbb916_2_690x108.png, https://aws1.discourse-cdn.com/github/optimized/2X/7/7c0fcbd8001398db393b9b9eb2e99af127fbb916_2_1035x162.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/7/7c0fcbd8001398db393b9b9eb2e99af127fbb916_2_1380x216.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7c0fcbd8001398db393b9b9eb2e99af127fbb916_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">registry_package.png</span><span class="informations">1469×231 16.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jeking3** [comment]
<p>Thanks, that’s good information.  Is that in the documentation?  </p>

**Comment by Yanjingzhu** [comment]
<p>There is the document:</p>
<p><a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#about-workflow-events" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#about-workflow-events </a></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/36b6ff8aff2cdc0de0820c03e4b8471ec0e09a18.png" data-download-href="/uploads/short-url/7O1R1F1cFpj0ZCeZhIVCG29Vaoo.png?dl=1" title="doc.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181092314-cc5987d0-097b-4b49-ad34-fa2f4d192423.png" alt="doc.png" data-base62-sha1="7O1R1F1cFpj0ZCeZhIVCG29Vaoo" width="690" height="325" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/36b6ff8aff2cdc0de0820c03e4b8471ec0e09a18_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">doc.png</span><span class="informations">708×334 32.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
